linux/network_device_ops/pvrdma_client_network_device_ops.yml

```diff
@@ -0,0 +1,138 @@
+          with_list: "{{ range(1, 11) | list }}"
+          loop_control:
+            loop_var: retry_count
+      when: >
+        (not pvrdma_client_vm_guest_ip) or
+        (pvrdma_client_vm_guest_ip == pvrdma_server_vm_guest_ip)
+
+    - name: "Check client VM's guest IP is not same with server VM's guest IP"
+      ansible.builtin.assert:
+        that:
+          - pvrdma_client_vm_guest_ip != pvrdma_server_vm_guest_ip
+        fail_msg: >-
+          Client VM '{{ pvrdma_client_vm_name }}' guest IP is '{{ pvrdma_client_vm_guest_ip }}',
+          which is same with server VM '{{ pvrdma_server_vm_name }}' guest IP after refreshing 10 times.
+        success_msg:
+          - "Server VM '{{ pvrdma_server_vm_name }}' guest IP is '{{ pvrdma_server_vm_guest_ip }}'."
+          - "Client VM '{{ pvrdma_client_vm_name }}' guest IP is '{{ pvrdma_client_vm_guest_ip }}'."
+
+    - name: "Wait for client VM being connectable"
+      include_tasks: ../../common/vm_wait_connection.yml
+      vars:
+        vm_wait_connection_timeout: 600
+
+    - name: "Add client VM's IP to in-memory inventory"
+      include_tasks: ../../common/add_host_in_memory_inventory.yml
+      vars:
+        add_host_in_memory_inventory_ip: "{{ vm_guest_ip }}"
+        add_host_in_memory_inventory_gp: "target_vm"
+        add_host_in_memory_inventory_user: "{{ vm_username }}"
+        add_host_in_memory_inventory_pwd: "{{ vm_password }}"
+        add_host_in_memory_inventory_python: "{{ vm_python | default('auto') }}"
+        add_host_in_memory_inventory_ssh_pipeline: "{{ vm_ssh_pipeline_enable | default(false) }}"
+
+    - name: "Display inventory hosts in group 'target_vm'"
+      ansible.builtin.debug: var=groups['target_vm']
+
+    - name: "Set fact of client VM is ready for PVRDMA testing"
+      ansible.builtin.set_fact:
+        pvrdma_client_vm_ready: true
+
+    - name: "Hot add a new {{ adapter_type }} network adapter on client VM and apply network config"
+      include_tasks: hot_add_network_adapter.yml
+      vars:
+        new_nic_ipv4: "{{ pvrdma_client_vm_ipv4 }}"
+
+    - name: "Check the result of pinging gateway from client VM"
+      ansible.builtin.assert:
+        that:
+          - ping_success_after_hotadd | bool
+        fail_msg: "Failed to ping gateway from PVRDMA device on client VM {{ vm_name }}"
+
+    - name: "Check PVRDMA device status and reload it if necessary on client VM"
+      include_tasks: check_and_reload_pvrdma.yml
+
+    - name: "Perform PVRDMA rping test from client VM to server VM"
+      include_tasks: pvrdma_rping_test.yml
+
+    - name: "Power off the client VM"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: 'powered-off'
+
+    - name: "Remove the client VM"
+      include_tasks: ../../common/vm_remove.yml
+
+    - name: "Set fact of test success on client VM"
+      ansible.builtin.set_fact:
+        pvrdma_client_test_success: true
```

Comment:
I mean if there is task failed in this block, then tasks in below `rescue` will be executed, at the end can add a `fail` task, then will go to the outside `rescue`, so this parameter 'pvrdma_client_test_success' is no need to be set here and be validated outside in `pvrdma_network_device_ops.yml` line 107.